src/DGtal/topology/tables/LookUpTableFunctions.ih

```diff
@@ -0,0 +1,200 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file LookUpTableFunctions.ih
+ *
+ * @author Pablo Hernandez-Cerdan. Institute of Fundamental Sciences.
+ * Massey University. Palmerston North, New Zealand
+ *
+ * @date 2016/03/25
+ *
+ * Implementation of header LookUplTableFunctions.h 
+ *
+ * This file is part of the DGtal library.
+ */
+
+#include "DGtal/kernel/SpaceND.h"
+#include "DGtal/kernel/domains/HyperRectDomain.h"
+
+namespace DGtal{
+  namespace lut {
+
+  template <typename TObject, typename TMap>
+  void
+  generateSimplicityTable(
+      const typename TObject::DigitalTopology & dt,
+      TMap & map )
+  {
+    typedef typename TObject::DigitalSet DigitalSet;
+    typedef typename TObject::Point Point;
+    typedef typename DigitalSet::Domain Domain;
+    typedef typename Domain::ConstIterator DomainConstIterator;
+
+    Point p1 = Point::diagonal( -1 );
+    Point p2 = Point::diagonal(  1 );
+    Point c = Point::diagonal( 0 );
+    Domain domain( p1, p2 );
+    DigitalSet shapeSet( domain );
+    TObject shape( dt, shapeSet );
+    unsigned int k = 0;
+    for ( DomainConstIterator it = domain.begin(); it != domain.end(); ++it )
+      if ( *it != c ) ++k;
+    ASSERT( ( k < 32 )
+        && "[generateSimplicityTable] number of configurations is too high." );
+    unsigned int nbCfg = 1 << k;
+    for ( unsigned int cfg = 0; cfg < nbCfg; ++cfg )
+    {
+      if ( ( cfg % 1000 ) == 0 )
```

Comment:
generateSimplicityTable is only used in examples `examples/topology/generateSimplicityTables2D.cpp` and `...Tables3D.cpp` to generate the files. I have found the progress bar useful, because the process can be slow in 3D. But of course, whatever you prefer, I didn't know you had verbosity flag at compile time. Your call!